### ISSUE TITLE
[Bug] nested spans font-sizes parsed into incorrect marks

### ISSUE THREAD

**Issue by qlp-mathias** [ProseMirror/prosemirror#1483]
**Note: I originally posted this as a Re-Mirror [issue](https://github.com/remirror/remirror/issues/2263) but was re-directed here.**

I'm dealing with some HTML created by a legacy rich text editor. This editor sometimes seems to have produced some unnecessary - albeit valid - nesting, which is not getting parsed correctly.

### Steps to reproduce

Given a Remirror instance with
- stringHandler: 'html'
- TextSizeExtension
- and the following HTML content

```html
<!-- Input -->
<p>
<span style="font-size: 18px;">
  Text in 18px
  <span style="font-size: 24px;">
    Text in 24px
    <span style="font-size: 18px;"><!-- this is sanitised, even though it's a valid new font-size -->
      Text in 18px
      <span style="font-size: 24px;"><!-- this is sanitised, but not noticeable in the end result because the parent 18px is sanitised -->
        Text in 24px
      </span>
    </span>
  </span>
</span>
</p>
```

This results in the following rendered HTML (Notice that the text in the red box should also have a font-size of 18px)
![Screenshot 2024-07-23 at 10 18 08](https://github.com/user-attachments/assets/8e4bf16a-e7fe-45d1-8b84-4431dd28177a)

It also seems to be scoped to font-size. If I instead replace the font-size style with, say, font-weight in a similar fashion it renders as expected.

**Comment by qlp-mathias** [comment]
I just noticed that this might be a duplicate of https://github.com/ProseMirror/prosemirror/issues/907

**Comment by zunsthy** [comment]
the same problem as you. i think it was resolved in prosemirror-model@1.22.3

**Comment by marijnh** [comment]
Yes, this should be fixed now. Are you still seeing it @zunsthy, or did you mean to say that it is resolved for you?

**Comment by zunsthy** [comment]
i mean it is resolved after 1.22.3  : )

**Comment by marijnh** [comment]
Wonderful!
